src/s3ql/backends/swift.py

```diff
@@ -387,24 +422,64 @@ def open_read(self, key):
                 self.conn.disconnect()
             raise
 
-        return ObjectR(key, resp, self, meta)
+        md5 = hashlib.md5()
+        fh.seek(off)
+        copyfh(self.conn, fh, update=md5.update)
 
-    def open_write(self, key, metadata=None, is_compressed=False):
-        """
-        The returned object will buffer all data and only start the upload
-        when its `close` method is called.
-        """
-        log.debug('started with %s', key)
+        if etag != md5.hexdigest():
+            log.warning('MD5 mismatch for %s: %s vs %s', key, etag, md5.hexdigest())
+            raise BadDigestError('BadDigest', 'ETag header does not agree with calculated MD5')
+
+        return meta
+
+    def write_fh(
+        self,
+        key: str,
+        fh: BinaryIO,
+        metadata: Optional[Dict[str, Any]] = None,
+        len_: Optional[int] = None,
+    ):
+        '''Upload *len_* bytes from *fh* under *key*.
+
+        The data will be read at the current offset. If *len_* is None, reads until the
+        end of the file.
+
+        If a temporary error (as defined by `is_temp_failure`) occurs, the operation is
+        retried. Returns the size of the resulting storage object.
+        '''
 
         if key.endswith(TEMP_SUFFIX):
             raise ValueError('Keys must not end with %s' % TEMP_SUFFIX)
 
+        off = fh.tell()
+        if len_ is None:
+            fh.seek(0, os.SEEK_END)
+            len_ = fh.tell()
+        return self._write_fh(key, fh, off, len_, metadata or {})
+
+    @retry
+    def _write_fh(self, key: str, fh: BinaryIO, off: int, len_: int, metadata: Dict[str, Any]):
+
         headers = CaseInsensitiveDict()
-        if metadata is None:
-            metadata = dict()
-        self._add_meta_headers(headers, metadata, chunksize=self.features.max_meta_len)
+        self._add_meta_headers(headers, metadata)
```

Comment:
why remove the `, chunksize=self.features.max_meta_len` from this call?

Reply:
Accident, thanks for spotting!